webanno-api-dao/src/main/java/de/tudarmstadt/ukp/clarin/webanno/api/dao/DocumentServiceImpl.java

```diff
@@ -593,25 +603,39 @@ public JCas readAnnotationCas(SourceDocument aDocument, User aUser)
 
     @Override
     @Transactional
-    public JCas readAnnotationCas(AnnotationDocument aAnnotationDocument)
-        throws IOException
+    public JCas readAnnotationCas(SourceDocument aDocument, String aUserName)
+            throws IOException
     {
-        Validate.notNull(aAnnotationDocument, "Annotation document must be specified");
-        
-        SourceDocument aDocument = aAnnotationDocument.getDocument();
-        String user = aAnnotationDocument.getUser();
-        
+        User user = userRepository.get(aUserName);
+
+        // Check if there is an annotation document entry in the database. If there is none,
+        // create one.
+        AnnotationDocument annotationDocument = createOrGetAnnotationDocument(aDocument, user);
```

Comment:
I believe we only want to retrieve the CAS here without the side-efffect of creating an annotation document for it.